### PR TITLE
Document Module#delegate_missing_to in the guides

### PR DESCRIPTION
**Pull request by gsamokovarov**
Added a small section for it in the `Active Support Core Extensions`
guide.

### CODE REVIEW

**Comment by gsamokovarov** [comment]
screen shot 2017-05-29 at 16 00 08

File: guides/source/active_support_core_extensions.md [hunk 1]
```diff
@@ -837,6 +839,55 @@ In the previous example the macro generates `avatar_size` rather than `size`.
...
+    @event.detail.person || @event.creator
+  end
+end
+```
+
+The target can be anything callable within the object, e.g. instance
+variables, methods, constants, etc.
+
+WARNING: The delegated method calls must be public on the target, otherwise
+they will raise `NoMethodError`.
```
Comment (matthewd) [inline guides/source/active_support_core_extensions.md]: This warning demonstrates that the behaviour doesn't match your manual implementation. :wink:

Really though, I think the full `method_missing`/`respond_to_missing?` illustration is too much unnecessary detail. It works for `delegate`, but isn't an efficient use of space in general. Let's ditch it, and just focus on usage.

Maybe also consider reusing the User/Profile example... "or you could just forward everything" seems a logical leap from "here's how to delegate 4 methods in one line"

Comment (gsamokovarov) [inline guides/source/active_support_core_extensions.md]: Sure, I will try that! I :)

**Comment by gsamokovarov** [comment]
screen shot 2017-05-30 at 11 03 49

@matthewd I have condensed the example to the one above. Are you okay with it?

File: guides/source/active_support_core_extensions.md [hunk 2]
```diff
@@ -837,6 +839,25 @@ In the previous example the macro generates `avatar_size` rather than `size`.
 
 NOTE: Defined in `active_support/core_ext/module/delegation.rb`
 
+#### `delegate_missing_to`
+
+Imagine you would like to delegate all of the `Profile` methods to the `User`
+object. Instead of listing every `Profile` method in a `delegate` call, you can
```
Comment (matthewd) [inline guides/source/active_support_core_extensions.md]: > delegate all of the `Profile` methods to the `User` object

That sounds around the wrong way -- though the exact opposite isn't right either.

'to' -> 'from' might be sufficient?

Comment (gsamokovarov) [inline guides/source/active_support_core_extensions.md]: Updated it to:

> Imagine you would like to delegate all methods from the `Profile` to the `User`
> object...

Does it feel better now?

Comment (matthewd) [inline guides/source/active_support_core_extensions.md]: No, because that still doesn't match what the code says: `User.delegate_missing_to :profile`

Comment (gsamokovarov) [inline guides/source/active_support_core_extensions.md]: So another take, this time trying to go with what the code says. 😅

> Imagine you would like to delegate everything missing from the `User` object,
> to the `Profile` one. The `delegate_missing_to` macro let's you implement this
> in a breeze:

Comment (matthewd) [inline guides/source/active_support_core_extensions.md]: ☝🏻 looks good

("lets" not "let's")

Comment (gsamokovarov) [inline guides/source/active_support_core_extensions.md]: ✅

File: guides/source/active_support_core_extensions.md [hunk 3]
```diff
@@ -837,6 +839,25 @@ In the previous example the macro generates `avatar_size` rather than `size`.
...
+  has_one :profile
+
+  delegate_missing_to :profile
+end
+```
+
+The target can be anything callable within the object, e.g. instance variables,
+methods, constants, etc.
+
+NOTE: Defined in `active_support/core_ext/module/delegation.rb`
```
Comment (vipulnsward) [inline guides/source/active_support_core_extensions.md]: lgtm. We should mention only public methods are delegated.

Comment (gsamokovarov) [inline guides/source/active_support_core_extensions.md]: Just dropped it in a sentence. If you think a warning, like the one in https://github.com/rails/rails/pull/29268#discussion_r118934489 works better, ping me and I'll change it.

**Pull request merged**